### PR TITLE
OFF-1457

### PR DESCRIPTION
**Pull request by MishelleBit**

### CODE REVIEW

File: adapters/flipp/flipp.go [hunk 1]
```diff
@@ -219,7 +219,32 @@ func buildBid(decision *InlineModel, impId string) *openrtb2.Bid {
 		if decision.Contents[0].Data.Width != 0 {
 			bid.W = decision.Contents[0].Data.Width
 		}
-		bid.H = 0
+		customDataInterface := decision.Contents[0].Data.CustomData
+		customDataMap, ok := customDataInterface.(map[string]interface{})
+		var defaultStandardHeight int64 = 2400
+		var defaultCompactHeight int64 = 600
```
Comment (hasan-kanjee) [inline adapters/flipp/flipp.go]: should move this to the top of the file with all the other consts

Comment (mike-lei) [inline adapters/flipp/flipp.go]: We should use `const` for constants and place them at the beginning of the file

File: adapters/flipp/flipp.go [hunk 2]
```diff
@@ -219,7 +219,32 @@ func buildBid(decision *InlineModel, impId string) *openrtb2.Bid {
...
+				if height, exists := customDataMap["standardHeight"].(int64); exists {
+					bid.H = height
+				} else {
+					bid.H = defaultStandardHeight
+				}
+			} else {
+				bid.H = defaultStandardHeight
+			}
+		}
 	}
```
Comment (hasan-kanjee) [inline adapters/flipp/flipp.go]: You can simplify this a little further by doing

```go
if flippExtParams.Options.StartCompact {
	defaultHeight = defaultCompactHeight
} else {
	defaultHeight = defaultStandardHeight
}

bid.H = defaultHeight

if ok { 
	if height, exists := customDataMap[flippExtParams.Options.StartCompact ? "compactHeight" : "standardHeight"].(int64); exists {
		bid.H = height
	}
}
```

**Review by hasan-kanjee — CHANGES_REQUESTED** [review]

File: adapters/flipp/flipp.go (hunk 2 above)
Comment (MishelleBit) [reply adapters/flipp/flipp.go]: > ```go
> 	if height, exists := customDataMap[flippExtParams.Options.StartCompact ? "compactHeight" : "standardHeight"].(int64); exists {
> ```

I get an error for the ? operator, saying **expected ']', found 'ILLEGAL'syntax**

Comment (hasan-kanjee) [reply adapters/flipp/flipp.go]: Sorry 🤦‍♂️ you can do this instead

```go
if flippExtParams.Options.StartCompact {
	defaultHeight = 600 // Default compact height
} else {
	defaultHeight = 2400 // Default standard height
}

// Set default height first
bid.H = defaultHeight

if ok { // customDataMap exists
	var key string
	if flippExtParams.Options.StartCompact {
		key = "compactHeight"
	} else {
		key = "standardHeight"
	}

	if height, exists := customDataMap[key].(int64); exists {
		bid.H = height
	}
}
```

**Pull request closed without merging**